File: Examples/python/i2c_eeprom_read_calib_125_14.py
#!/usr/bin/python3

import sys
import time
import struct
import redpitaya_scpi as scpi

rp_s = scpi.scpi(sys.argv[1])

rp_s.tx_txt('I2C:DEV80 "/dev/i2c-0"')
print("Init I2C")

rp_s.tx_txt('I2C:DEV?')
print(f"Device address: {rp_s.rx_txt()}")

rp_s.tx_txt('I2C:FMODE ON')
print("Set force mode")
rp_s.check_error()
# Eeprom 24c64 supports reading only 32 bytes of data at a time and only works through IOCTL

# set read address = 0
rp_s.tx_txt('I2C:IO:W:B2 0,0')
print("Write address for read")

rp_s.tx_txt('I2C:IO:R:B32?')
b1 = rp_s.rx_txt().strip('{').strip('}')

rp_s.tx_txt('I2C:IO:R:B16?')
b2 = rp_s.rx_txt().strip('{').strip('}')
rp_s.check_error()

buff = (b1 + "," + b2).split(",")
byte_array = bytearray(b'')
for s in buff:
    byte_array.append(int(s))

calib = [struct.unpack('i', byte_array[i:i+4])[0] for i in range(0, len(byte_array), 4)]
calibId, wp, res1, res2 = (calib[0] & 0xFFFFFFFF).to_bytes(4, 'little')

if (calibId == 1):
    print("ADC Ch1 High", calib[2])
    print("ADC Ch2 High", calib[3])
    print("ADC Ch1 Low", calib[4])
    print("ADC Ch2 Low", calib[5])
    print("ADC Ch1 Low offset", calib[6])
    print("ADC Ch2 Low offset", calib[7])
    print("DAC Ch1", calib[8])
    print("DAC Ch2", calib[9])
    print("DAC Ch1 offset", calib[10])
    print("DAC Ch2 offset", calib[11])
else:
    print("Unknown format of calibration. recived data:",byte_array)

File: Examples/Matlab/i2c_eeprom_read_calib_125_14.m
%% Define Red Pitaya as TCP/IP object

IP= '';           % Input IP of your Red Pitaya...
port = 5000;
tcpipObj=tcpip(IP, port);

%% Open connection with your Red Pitaya

fopen(tcpipObj);
tcpipObj.Terminator = 'CR/LF';
fprintf(tcpipObj,'I2C:DEV80 "/dev/i2c-0"');

fprintf(tcpipObj,'I2C:FMODE ON');          % set force mode

% Eeprom 24c64 supports reading only 32 bytes of data at a time and only works through IOCTL

fprintf(tcpipObj,'I2C:IO:W:B2 0,0'); % set read address = 0


b1 = query(tcpipObj,'I2C:IO:R:B32'); % read 32 bytes from iic 

b2 = query(tcpipObj,'I2C:IO:R:B16'); % read 16 bytes from iic

b_num = str2num(b1(1,2:length(b1)-3));

b_num(33:48) = str2num(b2(1,2:length(b2)-3));

calib = typecast(uint8(b_num),'int32');

fprintf('ADC Ch1 High %d\n',calib(3));
fprintf('ADC Ch2 High %d\n',calib(4));
fprintf('ADC Ch1 Low %d\n',calib(5));
fprintf('ADC Ch2 Low %d\n',calib(6));
fprintf('ADC Ch1 Low offset %d\n',calib(7));
fprintf('ADC Ch2 Low offset %d\n',calib(8));
fprintf('DAC Ch1 %d\n',calib(9));
fprintf('DAC Ch2 %d\n',calib(10));
fprintf('DAC Ch1 offset %d\n',calib(11));
fprintf('DAC Ch2 offset %d\n',calib(12));

%% Close connection with Red Pitaya

fclose(tcpipObj);